app/controllers/clearance/passwords_controller.rb

```diff
@@ -24,10 +27,12 @@ def new
   end
 
   def update
-    @user = find_user_for_update
+    @password_reset = find_password_reset_by_user_id_and_token
+    user = @password_reset.user
 
-    if @user.update_password password_reset_params
-      sign_in @user
+    if user.update_password password_reset_params
+      Clearance::PasswordResetDeactivator.new(user).run
```

Comment:
I was thinking of a scenario where the password gets updated but then for whatever reason the deactivator errors out. In that case, their password would be changed but their token would still be valid. To get around that, you can:

```ruby
ActiveRecord::Base.transaction do
  user.update_password password_reset_params
  deactivator.run
  true
end
```

In that way, if any of those statements fail, the database changes are rolled back.